app/services/concerns/full_sync_error_handler.rb

```diff
@@ -0,0 +1,7 @@
+module FullSyncErrorHandler
+  def raise_update_error(updates = {})
+    return if updates.none?
+
+    raise TeacherTrainingPublicAPI::FullSyncUpdateError, "#{updates.keys.to_sentence} have been updated"
```

Comment:
Discussed in person: to avoid the `Sidekiq` job being marked as failed maybe we can instead send the exception straight to Sentry `Sentry.capture_exception`